src/features/scheduling/generation/BunkJamboreeScheduler.ts

```
@@ -1,31 +1,389 @@
+      
+      const unassignedBunks = new Set<number>(this.bunks.map(b => b.id));
+
+      const MAX_CAPACITY_BUNKS = Math.floor(this.bunks.length/activities.length);
+
+      for (let i = 1; i <= activities.length; i++) {
+
+        for (const activity of activities) {
+
+
+          // Find bunks that have the activity as ith priority
+          let bunkPrefs = this.bunks.filter(b => this.preferences[blockID][b.id][activity.name] == i);
```

ESLint

[error] 'bunkPrefs' is never reassigned. Use 'const' instead.

(prefer-const)